### PR TITLE
Add user servlet

### PR DESCRIPTION
**Pull request by jyx11011**
Add get method in user servlet
Add json formatter for user

### CODE REVIEW

File: src/main/java/com/google/step/servlets/UserServlet.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+public class UserServlet extends HttpServlet {
+
+  private UserManager userManager = new UserManagerDatastore();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+
+    String idString = request.getPathInfo().substring(1); // Remove '/'
+    long id = Long.parseLong(idString);
```
Comment (limli) [inline src/main/java/com/google/step/servlets/UserServlet.java]: might need to try catch NumberFormatException if the parsing fails

File: src/main/java/com/google/step/servlets/UserServlet.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    List<Tag> tags = new ArrayList<>(); // followManager.getTagsFollowedByUser(id);
+    List<Restaurant> restaurants =
+        new ArrayList<>(); // followManager.getRestaurantsFollowedByUser(id);
+
+    String json = JsonFormatter.getUserJson(user, deals, following, followers, tags, restaurants);
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (limli) [inline src/main/java/com/google/step/servlets/UserServlet.java]: what does this POST method do?

File: src/main/java/com/google/step/servlets/UserServlet.java (hunk 1 above)
Comment (jyx11011) [reply src/main/java/com/google/step/servlets/UserServlet.java]: oh I do this in the next PR. I'll move it here.

File: src/main/java/com/google/step/servlets/UserServlet.java (hunk 2 above)
Comment (jyx11011) [reply src/main/java/com/google/step/servlets/UserServlet.java]: I haven't implemented this.

File: src/main/java/com/google/step/servlets/UserServlet.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+/** Servlet that handles users. */
+@WebServlet("/api/users/*")
+public class UserServlet extends HttpServlet {
+
+  private UserManager userManager = new UserManagerDatastore();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
```
Comment (anuragshirolkar) [inline src/main/java/com/google/step/servlets/UserServlet.java]: Can you move it down closer to where the actual response json object is written?

File: src/main/webapp/js/user-page.js [hunk 4]
```diff
@@ -131,11 +14,11 @@ function configureUserProfile(user) {
...
-  configureDealsPublishedBy(user);
-  configureUserFollowers(user);
-  configureUsersFollowedBy(user);
-  configureRestaurantsFollowedBy(user);
-  configureTagsFollowedBy(user);
+  //configureDealsPublishedBy(user);
+  //configureUserFollowers(user);
+  //configureUsersFollowedBy(user);
+  //configureRestaurantsFollowedBy(user);
+  //configureTagsFollowedBy(user);
```
Comment (anuragshirolkar) [inline src/main/webapp/js/user-page.js]: Did you mean to remove this?

**Review by anuragshirolkar — APPROVED** [review]

**Review by anuragshirolkar — DISMISSED** [review]

**Review by anuragshirolkar — APPROVED** [review]

**Pull request merged**